File: dev/update_fe7_affinities.js
const fs = require('fs');
const unitsPath = 'data/blazing_blade/units.json';
const scrapedPath = 'scraped_fe7.json';

const unitsData = JSON.parse(fs.readFileSync(unitsPath, 'utf8'));
const scrapedData = JSON.parse(fs.readFileSync(scrapedPath, 'utf8'));

let modified = false;

for (const unit of unitsData) {
    let searchName = unit.name;
    if (searchName.endsWith(' (FE7)')) {
        searchName = searchName.replace(' (FE7)', '');
    }

    let scrapedUnit = scrapedData.bases[searchName];

    // Special cases if any
    if (!scrapedUnit && searchName === 'Wallace') {
        scrapedUnit = scrapedData.bases['Wallace *'] || scrapedData.bases['Wallace'];
    }

    if (scrapedUnit && scrapedUnit.affinity) {
        if (unit.affinity !== scrapedUnit.affinity) {
            unit.affinity = scrapedUnit.affinity;
            console.log(`Updated affinity for ${unit.name} to ${unit.affinity}`);
            modified = true;
        }
    } else {
        console.log(`Could not find scraped affinity for ${unit.name}`);
    }
}

if (modified) {
    fs.writeFileSync(unitsPath, JSON.stringify(unitsData, null, 4));
    console.log('Successfully updated affinities.');
} else {
    console.log('No affinities needed updating.');
}

File: dev/update_fe7_affinities.py
import json
import os

units_file = 'data/blazing_blade/units.json'
scraped_file = 'scraped_fe7.json'

with open(units_file, 'r', encoding='utf-8') as f:
    units_data = json.load(f)
    
with open(scraped_file, 'r', encoding='utf-8') as f:
    scraped_data = json.load(f)
    
bases = scraped_data.get('bases', {})

modified = False
for unit in units_data:
    name = unit.get('name', '')
    search_name = name
    if search_name.endswith(' (FE7)'):
        search_name = search_name.replace(' (FE7)', '')
        
    scraped_unit = bases.get(search_name)
    if not scraped_unit and search_name == 'Wallace':
        scraped_unit = bases.get('Wallace *') or bases.get('Wallace')
        
    if scraped_unit and scraped_unit.get('affinity'):
        # Filter affinities to exact known types or valid strings
        # E.g. Affin_Anima might be the image name if alt was empty, but usually alt="Anima"
        affin = scraped_unit['affinity']
        if affin.startswith('Affin_'):
            affin = affin.replace('Affin_', '').replace('.png', '')
            
        if unit.get('affinity') != affin:
            unit['affinity'] = affin
            modified = True
            print(f"Updated affinity for {name} to {affin}")
    else:
        print(f"Could not find valid affinity for {name}")

if modified:
    with open(units_file, 'w', encoding='utf-8') as f:
        json.dump(units_data, f, indent=4)
    print("Done writing.")
else:
    print("No changes made.")
